src/udp_wrap.cc

```diff
@@ -397,6 +401,43 @@ void UDPWrap::DropMembership(const FunctionCallbackInfo<Value>& args) {
   SetMembership(args, UV_LEAVE_GROUP);
 }
 
+void UDPWrap::SetSourceMembership(const FunctionCallbackInfo<Value>& args,
+                                  uv_membership membership) {
+  UDPWrap* wrap;
+  ASSIGN_OR_RETURN_UNWRAP(&wrap,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EBADF));
+
+  CHECK_EQ(args.Length(), 3);
+
+  node::Utf8Value source_address(args.GetIsolate(), args[0]);
+  node::Utf8Value group_address(args.GetIsolate(), args[1]);
+  node::Utf8Value iface(args.GetIsolate(), args[2]);
+
+  const char* iface_cstr = *iface;
```

Comment:
I would also add `if (*iface == nullptr) return;`. We don’t have an explicit typecheck for `iface`, so converting it to string could throw an exception, and it would imo be odd to have this function throw an exception but still perform the call to `uv_udp_set_source_membership` in that case